models/models.go

```diff
@@ -127,13 +127,65 @@ func SetEngine() (err error) {
 	return nil
 }
 
+func migrateAccessToCollaboration() error {
+	sql := "select * from access"
+	results, err := x.Query(sql)
+	if err != nil {
+		return err
+	}
+
+	for _, result := range results {
+		userName := string(result["user_name"])
+		repoName := string(result["repo_name"])
+
+		user, err1 := GetUserByName(userName)
+		repo, err2 := GetRepositoryByRef(repoName)
+		if err1 != nil || err2 != nil {
+			continue
+		}
+		if err := repo.GetOwner(); err != nil {
+			continue
+		}
+		if repo.Owner.Id == user.Id {
+			continue
+		}
+
+		if repo.Owner.IsOrganization() {
+			if err := repo.Owner.GetMembers(); err != nil {
+				continue
+			}
+			if !repo.Owner.IsOrgMember(user.Id) {
+				x.Insert(&Collaboration{UserId: user.Id, RepoId: repo.Id})
+			}
+		} else {
+			x.Insert(&Collaboration{UserId: user.Id, RepoId: repo.Id})
+		}
+	}
+	return nil
+}
+
 func NewEngine() (err error) {
 	if err = SetEngine(); err != nil {
 		return err
 	}
+
+	var hasAccess, hasCollaboration, needMigration bool
+	if hasAccess, err = x.IsTableExist("access"); err != nil {
```

Comment:
Good point. However, we can't put the two block together, as we want to test for the collaboration table before the Sync, and do the actual filling after the table has been synced.